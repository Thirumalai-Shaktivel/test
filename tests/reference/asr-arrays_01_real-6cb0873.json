{
    "basename": "asr-arrays_01_real-6cb0873",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/arrays_01_real.f90",
    "infile_hash": "52d4b470c5523650dead5fa570d62db7293b9ecd23758e3cc4ba28c2",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-arrays_01_real-6cb0873.stdout",
    "stdout_hash": "84d7ee9bb0ae8fb01568e9af2cf69f6f289da5a89b94e106e245acf9",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}