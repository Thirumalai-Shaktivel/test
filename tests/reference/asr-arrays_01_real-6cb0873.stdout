(TranslationUnit (SymbolTable 1 {arrays_01_real: (Program (SymbolTable 2 {a: (Variable 2 a Local () Default (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))]) Source Public), b: (Variable 2 b Local () Default (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))]) Source Public), c: (Variable 2 c Local () Default (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))]) Source Public), i: (Variable 2 i Local () Default (Integer 4 []) Source Public), j: (Variable 2 j Local () Default (Integer 4 []) Source Public)}) arrays_01_real [] [(DoLoop ((Var 2 i) (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])) ()) [(= (ArrayRef 2 a [(() (Var 2 i) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (ImplicitCast (BinOp (Var 2 i) Add (ConstantInteger 10 (Integer 4 [])) (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])))]) (If (Compare (ArrayRef 2 a [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 11 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 a [(() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 12 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 a [(() (ConstantInteger 3 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 13 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (DoLoop ((Var 2 i) (ConstantInteger 11 (Integer 4 [])) (ConstantInteger 14 (Integer 4 [])) ()) [(= (ArrayRef 2 b [(() (BinOp (Var 2 i) Sub (ConstantInteger 10 (Integer 4 [])) (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (ImplicitCast (Var 2 i) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])))]) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 11 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 12 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 3 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 13 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 4 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 14 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (DoLoop ((Var 2 i) (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])) ()) [(= (ArrayRef 2 b [(() (Var 2 i) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (BinOp (ArrayRef 2 a [(() (Var 2 i) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) Sub (ImplicitCast (ConstantInteger 10 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])))]) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 1 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 2 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 3 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 3 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (= (ArrayRef 2 b [(() (ConstantInteger 4 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (BinOp (BinOp (BinOp (ArrayRef 2 b [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) Add (ArrayRef 2 b [(() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) Add (ArrayRef 2 b [(() (ConstantInteger 3 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) Add (ArrayRef 2 a [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))])) (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))]))) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 4 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 17 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (= (ArrayRef 2 b [(() (ConstantInteger 4 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (ArrayRef 2 a [(() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 3 (Integer 4 [])))]))) (If (Compare (ArrayRef 2 b [(() (ConstantInteger 4 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 11 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 4 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (DoLoop ((Var 2 i) (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])) ()) [(DoLoop ((Var 2 j) (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])) ()) [(= (ArrayRef 2 c [(() (Var 2 i) ()) (() (Var 2 j) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) (ImplicitCast (BinOp (BinOp (Var 2 i) Add (Var 2 j) (Integer 4 [])) Add (ConstantInteger 10 (Integer 4 [])) (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])))])]) (If (Compare (ArrayRef 2 c [(() (ConstantInteger 1 (Integer 4 [])) ()) (() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 12 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 c [(() (ConstantInteger 1 (Integer 4 [])) ()) (() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 13 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 c [(() (ConstantInteger 2 (Integer 4 [])) ()) (() (ConstantInteger 1 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 13 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] []) (If (Compare (ArrayRef 2 c [(() (ConstantInteger 2 (Integer 4 [])) ()) (() (ConstantInteger 2 (Integer 4 [])) ())] (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) NotEq (ImplicitCast (ConstantInteger 14 (Integer 4 [])) IntegerToReal (Real 8 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 []))) ((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))])) (Logical 4 [])) [(ErrorStop ())] [])])}) [])
